tests/unit/widget/widget_classes.js

```diff
@@ -0,0 +1,131 @@
+(function( $ ) {
+
+module( "widget factory classes", {
+	teardown: function() {
+		if ( $.ui ) {
+			delete $.ui.classWidget;
+			delete $.fn.classWidget;
+		}
+	}
+});
+
+$.widget( "ui.classesWidget", {
+	options: {
+		classes: {
+			"test-wrapper": "self-wrapper",
+			"test-self": "self-class self-class-2"
```

Comment:
The `test` prefix isn't used consistently, as far as I can tell. It would help to use a prefix for structure classes (like "ui-tooltip-content") and another one for styling classes (like "ui-corner-all"). Would make it much easier to follow these tests.